curvefs/src/metaserver/s3compact_wq_impl.h

```diff
@@ -0,0 +1,122 @@
+/*
+ *  Copyright (c) 2021 NetEase Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/*
+ * @Project: curve
+ * @Date: 2021-09-07
+ * @Author: majie1
+ */
+
+#ifndef CURVEFS_SRC_METASERVER_S3COMPACT_WQ_IMPL_H_
+#define CURVEFS_SRC_METASERVER_S3COMPACT_WQ_IMPL_H_
+
+#include <algorithm>
+#include <condition_variable>
+#include <deque>
+#include <list>
+#include <memory>
+#include <mutex>
+#include <string>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+#include "curvefs/src/metaserver/s3compact_wq.h"
+
+using ::curve::common::Configuration;
+using ::curve::common::InitS3AdaptorOption;
+using ::curve::common::S3Adapter;
+using ::curve::common::S3AdapterOption;
+using ::curve::common::TaskThreadPool;
+
+namespace curvefs {
+namespace metaserver {
+
+class S3CompactWorkQueueImpl : public TaskThreadPool<> {
+ public:
```

Comment:
all members are public ?

Reply:
yes, all public in S3CompactWorkQueueImpl. S3CompactWorkQueue/S3CompactWorkQueueImpl are writted using PIMPL idiom.